src/graph/delta_matrix/delta_extract.c

```diff
@@ -5,36 +5,46 @@
 
 #include "RG.h"
 #include "delta_matrix.h"
+#define DM_extractElement(TYPE_SUFFIX, CTYPE)                                  \
+GrB_Info Delta_Matrix_extractElement_##TYPE_SUFFIX                             \
+(                                                                              \
+	CTYPE *x,              /* extracted scalar                */               \
+	const Delta_Matrix A,  /* matrix to extract a scalar from */               \
+	GrB_Index i,           /* row index                       */               \
+	GrB_Index j            /* column index                    */               \
+) {                                                                            \
+	/* validate */                                                             \
+	ASSERT(x != NULL);                                                         \
+	ASSERT(A != NULL);                                                         \
+	GrB_Info info;                                                             \
+	GrB_Matrix m  = DELTA_MATRIX_M(A);                                         \
+	GrB_Matrix dp = DELTA_MATRIX_DELTA_PLUS(A);                                \
+	GrB_Matrix dm = DELTA_MATRIX_DELTA_MINUS(A);                               \
+                                                                               \
+	GrB_Type ty = NULL;                                                        \
+	GrB_OK(GxB_Matrix_type(&ty, m));                                           \
+	ASSERT(ty == GrB_##TYPE_SUFFIX);                                           \
+                                                                               \
+	/* if dp[i,j] exists return it */                                          \
+	GrB_OK (info = GrB_Matrix_extractElement(x, dp, i, j));                    \
+	if(info == GrB_SUCCESS) {                                                  \
+		return GrB_SUCCESS;                                                    \
+	}                                                                          \
```

Comment:
I think we should start checking `M` as it's much more likely to find the entry there.